modules/openy_gc_shared_content_server/src/Entity/SharedContentSource.php

```diff
@@ -131,6 +131,37 @@ public function setStatus($status) {
     return $this;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getApiUpdated() {
+    return (bool) $this->get('api_updated')->value;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function isUpdated() {
+    $url = $this->getUrl();
+
+    // If there's no url then we're done.
+    if (empty($url)) {
+      return FALSE;
+    }
+
+    // Attempt to hit the new endpoint and get the status code.
+    // Without the proper headers it will still return 200, but no content.
+    $url .= '/api/virtual-y/shared-content-source/gc_video';
+    $client = \Drupal::httpClient();
+    $status = $client->get($url, ['http_errors' => FALSE])->getStatusCode();
```

Comment:
I fear that report page could load for a long time (since it will check all sources during page load).
Can we try to run it in the background?

Reply:
@anpolimus right... I made an attempt at using `#lazy_builder` for this and I made some progress but I'm still having problems getting it to run async. It seemed like that method with BigPipe might be a good solution here but I've never done custom work with that before. I grabbed some time to discuss tomorrow.